src/controllers/ContactController.php

```diff
@@ -32,20 +60,26 @@ public function contact(Request $request, PDO $db)
         // only trusted clients can contact us to save on spam
         $clientId         = $request->getParameter('client_id');
         $clientSecret     = $request->getParameter('client_secret');
-        $this->oauthModel = $request->getOauthModel($db);
-        if (! $this->oauthModel->isClientPermittedPasswordGrant($clientId, $clientSecret)) {
+        $oauthModel = $request->getOauthModel($db);
+        if (! $oauthModel->isClientPermittedPasswordGrant($clientId, $clientSecret)) {
             throw new Exception("This client cannot perform this action", 403);
         }
 
+        if (! isset($this->emailService)) {
```

Comment:
Don't need this anymore since we're constructor-injected :)

Reply:
done